chainer/variable.py

```diff
@@ -1088,8 +1088,9 @@ def set_grad(node, value):
                 in_grad.append(gx)
             in_grad = tuple(in_grad)
 
-            gxs = func.backward_accumulate(
-                target_input_indexes, out_grad, in_grad)
+            with cuda.get_device_from_array(*out_grad_data):
```

Comment:
I think it's better to fix this line, rather than adding with block here.
https://github.com/chainer/chainer/pull/4725/files#diff-e131a8dfc2ea641e7e7c63597b9b0d8eR1054